fetch.bs

```diff
@@ -8380,6 +8380,56 @@ correctly. This section aims to give some advice.
 
 <p class=XXX>This is a work in progress.
 
+<h3 id=fetch-elsewhere-request>Constructing a request</h3>
```

Comment:
A high-level comment for the whole section, which I've elaborated on in several of the specific comments:

I think we want this section to be instructions to the authors of calling specifications -- a how-to guide. It might make sense to also improve the descriptive text in https://fetch.spec.whatwg.org/#requests, but that's not the primary hole this section needs to fill.

Reply:
Agree that it would be good to format this more as a how-to, and in particular have more bullet lists (and mandatory parameters first and common optional parameters after). A hard part about using fetch is how to invoke it and continue processing the response after it's done. Based on the title of the section it would be out of scope but wanted to point out the gap there as well. There are not many good examples of using fetch in a nontrivial way where the result is 'awaited' by the algorithm, which then does something with the response.

Reply:
I focused on requests first because the callback part is already kinda covered [in the subsequent bit](https://fetch.spec.whatwg.org/#fetch-elsewhere-fetch). Is something missing in that section?

Reply:
I think that subsequent section could use an example of how to use it. I think a lot of people expect to be able to say: 
1. In parallel:
   1. fetch the request.
   2. Wait for the fetch to finish.
   3. Use the response.

And with the setup of the *processResponse* algorithms, that just doesn't work. You have to do something more like

1. Fetch the request, with the following algorithms as arguments:
   <dl><dt>processResponse<dd>Steps
   <dt>processResponseConsumeBody<dd>Other Steps

It might also be helpful to rewrite that section to map from a spec's goals about how it'll handle the response, to the set of algorithm arguments it ought to pass.

But +100 for doing all that in a separate PR. Writing this section is a great first step.

Reply:
Gotcha, this makes sense. I will get to improving the callback section once we finish with the request section.
I also intend to add a section about the different cross-origin protections (CORS, TAO, CORB, CSP) with a short guide on which to use for what purpose.